src/benchmarks/gc/GC.Infrastructure/Notebooks/Reports.dib

```diff
@@ -588,21 +588,169 @@ public static class Metrics
         public static Metric<TraceGC> HeapCount = new(gc => gc.HeapCount, "Heap count", "#");
 
         // HeapCountSample
-        public static Metric<TraceGC> HcsElapsedTimeBetweenGCs = new(gc => gc.HeapCountSample.ElapsedTimeBetweenGCsMSec, "HCSampleElapsed", "ms");
-        public static Metric<TraceGC> HcsGCIndex = new(gc => gc.HeapCountSample.GCIndex, "HCSampleGCIndex", "#");
-        public static Metric<TraceGC> HcsGCPauseTime = new(gc => gc.HeapCountSample.GCPauseTimeMSec, "HCSampleGCPause", "ms");
-        public static Metric<TraceGC> HcsMslWaitTime = new(gc => gc.HeapCountSample.MslWaitTimeMSec, "HCSampleGCMslWait", "ms");
+        public static Metric<TraceGC> TCPToConsider = new(gc => 
```

Comment:
`gc => { return E; }` can be simplified as `gc => E`

In looking at the usage side, consider making `DynamicEvents` a property.  It "stands out" compared to the non-dynamic events here.

How does a dynamic event have a built-in property name (like `SizeAdapatationTuning`)?  I guess that's from an earlier PR or I missed it in this one.  So would a non-built-in one need to do a dictionary lookup of some sort?  Or use an extension method to wrap that and make it look "nice"?

Reply:
We can't make `DynamicEvents` a property - we have C# extension method, but not C# extension property, that's a language constraint.

Reply:
The high level goal of the dynamic event work is to make it really dynamic (i.e. as fluid as possible to introduce new event or change them) without having to involve changes in PerfView. Exposing the `byte[]` at the PerfView level abstraction and have the parsing done completely on the GC infrastructure side should future proof it.

We certainly could put all these parsing logic into PerfView, but then what are the trade offs? 

If we put the generic parsing into TraceEvent:

**Pros:**

1. We can use DynamicEvent as a property
2. We can store things on the TraceGC object (e.g. the parsed event objects)

**Cons:**

1. Next time when we want to do things in a new way again, we need to change PerfView again.
2. Any bug in TraceEvent needs some time to get in because of Nuget package publishing.

In fact, in the not too long future, I envision we will have to deal with variable-length array somehow (e.g. exposing some per heap data), so this will change again.

IMO, we have made the right trade-off. The small advantages doesn't worth the risk.

Alternatively, I could have made a `DynamicEvent` property on `TraceGC` that is a trivial read/write field of type `dynamic`. The parser on the API side can populate it and client can access it without going through extension method. This can be an option.